### ISSUE TITLE
PowerVR Vulkan bugs (Honor 7S)

### ISSUE THREAD

**Issue by hrydgard** [hrydgard/ppsspp#11743]
There's a bad assumption in VulkanDeviceAllocator that all objects with DEVICE_LOCAL_BIT will want to live in the same MemoryType. This is not the case on PowerVR, causing font textures and many other things to fail allocating. Check out the memory types: https://vulkan.gpuinfo.org/displayreport.php?id=4534#memory - there are duplicates, who knows how the driver picks which ones a certain image want to use. Maybe large textures and rendertargets in the first and small textures in the second?

Anyway, proposed solutions:

    * Tag each slab with memorytype, check when searching for a slab to allocate from
    * Switch to the battle-tested https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator

Additionally, it has that weird display stride problem causing the entire display to be corrupted. I suspect it's some bug in the driver triggered by creation order or something, will try some Android vulkan samples and see what they do to get a working picture.

**Comment by hrydgard** [comment]
That takes care of the first bug. Now just need to figure out why the swapchain is completely broken...

**Comment by hrydgard** [comment]
Turns out it's broken even when running Google's samples. Sigh. I've checked every parameter and carefully logged and nothing seems out of the ordinary.

I think I'll give up on Vulkan on these, by blacklisting PowerVR from using Vulkan entirely.

**Comment by hrydgard** [comment]
Turns out this one is solved by forcibly making the swapchain width a multiple of 32. This at least makes the menu usable, so I will commit it soon. However, we should definitely not default to Vulkan on these devices - there are several major rendering problems that show up in-game.

**Comment by unknownbrackets** [comment]
Note: we don't default to Vulkan on these devices (or any others) in the latest git.

-[Unknown]

**Comment by hrydgard** [comment]
True, but it's important that the UI is usable in case someone switches to it and wants to switch back.
